File: Clase 18/PYTHON/ejemplo_animales/Gato.py
from Animal import Animal

class Gato(Animal):

    def __init__(self, id, nombre, especie, tipo_pelaje) -> None:
        super().__init__(id, nombre, especie)
        self.tipo_pelaje = tipo_pelaje
    
    def hace_miau(self):
        print(f"{self.nombre} hace MIAUUU!!!!")
    
    # Polimorfismo de Sobrecarga
    def se_lava(self):
        print(f"{self.nombre} se lava con su lengua...")
    
    # Polimorfismo por inclusion
    def hace_sonido(self):
        self.hace_miau()

File: Clase 18/JAVA/ejemploAnimales/Gato.java
package ejemploAnimales;

public class Gato extends Animal{
    protected String tipoPelaje;

    public Gato(int id, String nombre, String especie, String tipoPelaje){
        super(id, nombre, especie);
        this.tipoPelaje = tipoPelaje;
    }

    public void haceMiau(){
        System.out.println(this.nombre + " hace MIUAUUU!!!!");
    }

    // Polimorfismo por sobrecarga
    public void seLava(){
        System.out.println(this.nombre + " se lava con la lengua...");
    }

    // Polimorfismo Por Inclusion
    @Override
    public void haceSonido(){
        this.haceMiau();
    }
}

